File: app/src/main/java/SQLData/WordRepository.java
package SQLData;

import android.app.Application;
import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.Observer;
import android.os.AsyncTask;
import android.support.annotation.Nullable;
import android.util.Log;

import com.example.andreip.myapplication.app.Recipe;

import java.util.List;

public class WordRepository {

    private DataAccessWord mWordDao;
    private LiveData<List<RecipeDb>> mAllWords;


    public WordRepository(Application application) {
        WordRoomDatabase db = WordRoomDatabase.getDatabase(application);
        mWordDao = db.wordDao();
        mAllWords = mWordDao.getAllWords();
    }

    public LiveData<List<RecipeDb>> getAllWords() {
        mAllWords.observeForever(new Observer<List<RecipeDb>>() {
            @Override
            public void onChanged(@Nullable List<RecipeDb> recipeDbs) {
                if (mAllWords.getValue().equals(null)) {
                    // get RetroFit
                } else
                    Log.d("Get Words", mAllWords.getValue().get(0).getTitle());
            }
        });

        return mAllWords;
    }

    public void insertAllRecipes(List<Recipe> recipes) {
        for (Recipe recipe : recipes)
            insert(new RecipeDb(recipe));
    }

    public void insert(RecipeDb recipe) {
        new insertAsyncTask(mWordDao).execute(recipe);
    }

    private static class insertAsyncTask extends AsyncTask<RecipeDb, Void, Void> {

        private DataAccessWord mAsyncTaskDao;

        insertAsyncTask(DataAccessWord dao) {
            mAsyncTaskDao = dao;
        }

        @Override
        protected Void doInBackground(final RecipeDb... params) {
            mAsyncTaskDao.insert(params[0]);
            return null;
        }
    }
}

File: app/src/main/java/SQLData/WordRepository.kt
package SQLData

import android.app.Application
import android.arch.lifecycle.LiveData
import android.os.AsyncTask
import android.util.Log

import com.example.andreip.myapplication.app.Recipe

import java.util.ArrayList

class WordRepository(application: Application) {

    private val mWordDao: DataAccessWord?
     val allWords: LiveData<List<RecipeDb>>


    init {
        val db = WordRoomDatabase.getDatabase(application)
        mWordDao = db?.wordDao()
        allWords = mWordDao!!.allWords
    }

    fun insertAllRecipes(recipes: List<Recipe>) {
        val recipeDbArrayList = ArrayList<RecipeDb>()
        Log.d("DEBUG", recipes.size.toString() + " ")
        for (recipe in recipes) {
            Log.d("DEBUG", recipe.getTitle()!! + " ")

            recipeDbArrayList.add(RecipeDb(recipe))
        }
        insert(recipeDbArrayList)
    }

    fun getValueAllWords() : List<RecipeDb>
    {
        return allWords.value!!
    }

    fun insert(recipe: List<RecipeDb>?) {

        if (recipe == null) {
            Log.d("ERROR", "No data to insert ")
            return
        }

        insertAsyncTask(mWordDao).execute(recipe)
    }

    private class insertAsyncTask internal constructor(private val mAsyncTaskDao: DataAccessWord?) : AsyncTask<List<RecipeDb>, Void, Void>() {

        override fun doInBackground(vararg params: List<RecipeDb>): Void? {

            mAsyncTaskDao?.deleteAll()
            for (recipeDb in params[0])
                mAsyncTaskDao?.insert(recipeDb)
            return null
        }
    }
}